File: MyApplication2/mylibrary/src/main/java/com/hld/mylibrary/TestActivity.java
package com.hld.mylibrary;

import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.alibaba.android.arouter.launcher.ARouter;

@Route(path = "/test/activity")
public class TestActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        TextView textView=new TextView(this);
        textView.setGravity(Gravity.CENTER);

        textView.setText("TestActivity");


        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ARouter.getInstance().build("/test2/activity").navigation();
            }
        });

        setContentView(textView);
    }
}

File: MyApplication2/app/src/main/java/com/hld/myapplication2/TestActivity.kt
package com.hld.myapplication2

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class TestActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(Button(this).apply {
            text="跳转"
            setOnClickListener {
                startActivity(Intent(this@TestActivity,TestActivity2::class.java))
            }
        })
    }
}